support/cas-server-support-inwebo-mfa/src/main/java/org/apereo/cas/support/inwebo/web/flow/actions/CheckUserAction.java

```diff
@@ -0,0 +1,79 @@
+import lombok.val;
+import org.springframework.context.MessageSource;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.webflow.action.AbstractAction;
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.execution.RequestContext;
+
+/**
+ * A web action to check the user (status).
+ *
+ * @author Jerome LELEU
+ * @since 6.4.0
+ */
+@RequiredArgsConstructor
+@Slf4j
+public class CheckUserAction extends AbstractAction implements WebflowConstants {
+
+    private final MessageSource messageSource;
+
+    private final InweboService service;
+
+    private final CasConfigurationProperties casProperties;
+
+    @Override
+    public Event doExecute(final RequestContext requestContext) {
+
+        val authentication = WebUtils.getInProgressAuthentication();
+        val login = authentication.getPrincipal().getId();
+        LOGGER.debug("Login: {}", login);
+
+        val flowScope = requestContext.getFlowScope();
+        val inwebo = casProperties.getAuthn().getMfa().getInwebo();
+        flowScope.put(SITE_ALIAS, inwebo.getSiteAlias());
+        flowScope.put(SITE_DESCRIPTION, inwebo.getSiteDescription());
+        flowScope.put(LOGIN, login);
+
+        try {
+            val response = service.loginSearch(login);
+            val oneUser = response.isOk() && response.getCount() == 1 && response.getUserId() > 0;
+            if (oneUser) {
+                val userIsBlocked = response.getUserStatus() == 1;
+                if (userIsBlocked) {
+                    LOGGER.error("User is blocked: {}", login);
+                    return error();
+                }
+                val activationStatus = response.getActivationStatus();
+                if (activationStatus == 0) {
+                    LOGGER.debug("User is not registered: {}", login);
+                    flowScope.put(MUST_ENROLL, true);
+                    flowScope.put(INWEBO_ERROR_MESSAGE, messageSource.getMessage("cas.inwebo.error.usernotregistered", null, LocaleContextHolder.getLocale()));
+                } else if (activationStatus == 1) {
+                    LOGGER.debug("User can only handle push notifications: {}", login);
+                    return getEventFactorySupport().event(this, PUSH);
+                } else if (activationStatus == 2) {
+                    LOGGER.debug("User can only handle browser authentication: {}", login);
+                    return getEventFactorySupport().event(this, BROWSER);
+                } else if (activationStatus == 3 || activationStatus == 5) {
+                    LOGGER.debug("User must select the authentication method: {}", login);
+                    return getEventFactorySupport().event(this, SELECT);
+                } else {
+                    LOGGER.error("Unknown activation status: {} for: {}", activationStatus, login);
+                }
+            } else {
+                LOGGER.error("No user found for: {}", login);
```

Comment:
All error calls to the logger should be using LoggingUtils.error()

Reply:
Yes, but only when there is an exception. I will change that.